starter-core/src/main/java/io/micronaut/starter/feature/server/HttpPoja.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.server;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.dependencies.Dependency;
+import io.micronaut.starter.options.BuildTool;
+import jakarta.inject.Singleton;
+
+@Singleton
+public class HttpPoja extends AbstractMicronautServerFeature {
+
+    @Override
+    public String getName() {
+        return "http-poja";
+    }
+
+    @Override
+    public String getTitle() {
+        return "Plain Old Java HTTP Application";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Add support for HTTP POJA based on Apache libraries";
+    }
+
+    @Override
+    public String getMicronautDocumentation() {
+        return "https://micronaut-projects.github.io/micronaut-servlet/latest/guide/index.html#http-poja";
+    }
+
+    @Override
+    public void doApply(GeneratorContext generatorContext) {
+        if (generatorContext.getBuildTool() == BuildTool.MAVEN) {
```

Comment:
why are the dependencies only applied for Maven builds?

Reply:
> why are the dependencies only applied for Maven builds?

I added a PR to the gradle plugin for another runtime: https://github.com/micronaut-projects/micronaut-gradle-plugin/pull/1024

I think that makes sense since it is a similar server implementation to other runtimes